### PR TITLE
Various CI updates

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

- remove Qiskit Aer (as it is not needed)
- ~~also test Python 3.11~~ (postponing this for now)
- bump to Python 3.8 where 3.7 is not necessary
- do not install Qiskit Terra from main (see also https://github.com/Qiskit/qiskit-nature/pull/1159)
- disable CI for forks (see also https://github.com/Qiskit/qiskit-nature/pull/1163)

### CODE REVIEW

**Review by Eric-Arellano — DISMISSED** [review]
Awesome! Thanks.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -177,18 +175,19 @@ jobs:
         run: |
           coverage3 combine
           mv .coverage ./ci-artifact-data/fin.dat
-        if: ${{ matrix.python-version == 3.7 }}
+        if: ${{ matrix.python-version == 3.8 }}
```
Comment (woodsp-ibm) [inline .github/workflows/main.yml:178]:
```suggestion
        if: ${{ matrix.os == 'ubuntu-latest' && matrix.python-version == 3.8 }}
```
Related to the `run-tests` change

File: setup.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2022.
+# (C) Copyright IBM 2022, 2023.
```
Comment (woodsp-ibm) [inline setup.py:3]: Not quite sure why this was updated just for the date - maybe the date tool did this

Comment (mrossinek) [reply setup.py:3]: I had done some other changes which I reverted later. This can be reverted, too :+1:
```suggestion
# (C) Copyright IBM 2022.
```

**Pull request merged**